decorator.rb

```diff
@@ -0,0 +1,29 @@
+require_relative './nameable'
+
+class Decorator < Nameable
+  attr_accessor :nameable
+
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    name_cap = @nameable.correct_name
+    name_cap.upcase
+  end
+end
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    name = @nameable.correct_name
+    output = name.slice(0, 10) if name.length > 10
+    output
+  end
```

Comment:
- [ ] Kindly add an else case for this method as if the name is less than 10 characters the method will return `nil`.